client/rpc/src/eth.rs

```diff
@@ -1386,6 +1620,16 @@ where
 				let mut cumulative_receipts = receipts.clone();
 				cumulative_receipts.truncate((status.transaction_index + 1) as usize);
 
+				let transaction = block.transactions[index].clone();
+				let effective_gas_price = match transaction {
+					EthereumTransaction::Legacy(t) => t.gas_price,
+					EthereumTransaction::EIP2930(t) => t.gas_price,
+					EthereumTransaction::EIP1559(t) => base_fee
+						.unwrap()
```

Comment:
Theoretically is not possible to have a EIP1559 receipt without a onchain base fee. However unwrap here is fishy. Match base_fee also and Error if is None in a 1559 context.